apps/files_sharing/src/views/SharingDetailsTab.vue

```diff
@@ -715,10 +703,23 @@ export default {
 					this.share.newPassword = await GeneratePassword()
 					this.advancedSectionAccordionExpanded = true
 				}
-				if (this.hasExpirationDate) {
-					this.share.expireDate = this.defaultExpiryDate
+				/* Set default expiration dates if set */
+				if (this.isPublicShare && this.config.isDefaultExpireDateEnabled) {
+					this.share.expireDate = this.config.defaultExpirationDate.toDateString()
+				}
+
+				if (this.isRemoteShare && this.config.isDefaultRemoteExpireDateEnabled) {
+					this.share.expireDate = this.config.defaultRemoteExpirationDateString.toDateString()
+				}
+
+				if (this.config.isDefaultInternalExpireDateEnabled) {
+					this.share.expireDate = this.config.defaultInternalExpirationDate.toDateString()
+				}
```

Comment:
```suggestion
				/* Set default expiration dates if set */
				if (this.isPublicShare && this.config.isDefaultExpireDateEnabled) {
					this.share.expireDate = this.config.defaultExpirationDate.toDateString()
				} else if (this.isRemoteShare && this.config.isDefaultRemoteExpireDateEnabled) {
					this.share.expireDate = this.config.defaultRemoteExpirationDateString.toDateString()
				} else if (this.config.isDefaultInternalExpireDateEnabled) {
					this.share.expireDate = this.config.defaultInternalExpirationDate.toDateString()
				}
```
This change fixes the behavior.